### PR TITLE
small Improvements to generate_code script

### PR DESCRIPTION
**Pull request by CoderDake**
This PR makes it so the generate_code script can be called from anywhere. 

# Changes
- The beginning of the script now automatically uses the `devtools` directory as an reference point.
- It also changes the permissions of the file so that it is executable by default.

## Screenshot
An example of the script being run from outside of the devtools directory
image

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Review by kenzieschmoll — APPROVED** [review]

File: tool/generate_code.sh [hunk 1]
```diff
@@ -3,15 +3,17 @@
 # Copyright 2022 The Chromium Authors. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
-
+TOOL_DIR=`dirname "${BASH_SOURCE[0]}"`
```
Comment (polina-c) [inline tool/generate_code.sh:6]: What is BASH_SOURCE[0]? Where it is documented?

Can you add explanation in a comment?

Comment (CoderDake) [reply tool/generate_code.sh]: BASH_SOURCE[0] essentially gives a path the the script, relative to where it was called from.

i've added some comments that hopefully help explain this :)

Comment (CoderDake) [reply tool/generate_code.sh]: https://stackoverflow.com/a/35006505/464770
Is where I learned about it :)

File: tool/generate_code.sh [hunk 2]
```diff
@@ -4,14 +4,25 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
 
+# Contains a path the this script, relative to the directory it was called from.
```
[the remark this answers is not in the document]
Comment (CoderDake) [reply tool/generate_code.sh:7]: oops ty

**Review by kenzieschmoll — APPROVED** [review]

**Pull request merged**